### PR TITLE
RHINENG-12334: filters for incidents

### PR DESCRIPTION
**Pull request by Fewwy**
Added functionality for filters 
I may need to fix things in the future when real incident data is present in the Prometheus database.
I added a JSdoc for the filter function, I found it much more useful than any typescript interfaces and other stuff, because it provides highlights and allows you to explain how the function works + provide an example

### CODE REVIEW

File: src/components/Incidents/IncidentsPage.jsx [hunk 1]
```diff
@@ -36,19 +41,20 @@ const IncidentsPage = ({ customDataSource, namespace }) => {
...
+    filter: (filter, incident) => filterIncident(incident, filter),
     filterGroupName: t('Incident type'),
     isMatch: () => console.log('here should be filter state'),
     items: [
       { id: 'long-standing', title: t('Long standing') },
       { id: 'informative', title: t('Informative') },
       { id: 'inactive', title: t('Inactive') },
     ],
-    type: 'name',
+    type: 'informative',
```
Comment (jgbernalp) [inline src/components/Incidents/IncidentsPage.jsx:52]: I think this type should reflect the filter grouping in this case should it be "incident-status" or something similar?

File: src/components/Incidents/utils.js [hunk 2]
```diff
@@ -11,8 +12,6 @@ export function groupAndDeduplicate(objects, keyType) {
       key = obj.metric.alertname + obj.metric.namespace;
     } else if (keyType === 'group_id') {
       key = obj.metric.group_id;
-    } else {
```
Comment (jgbernalp) [inline src/components/Incidents/utils.js:14]: If this is removed, we would create a new entry with the undefined key for the groupedObjects Map

Comment (Fewwy) [reply src/components/Incidents/utils.js:14]: This is not an issue, there are 2 places where this function is called. processAlertTimestamps and processIncidentTimestamps and in both cases we directly pass the string parameter.

File: src/components/Incidents/IncidentsPage.jsx (hunk 1 above)
Comment (Fewwy) [reply src/components/Incidents/IncidentsPage.jsx:52]: I will update the type

File: src/components/Incidents/utils.js (hunk 2 above)
Comment (jgbernalp) [reply src/components/Incidents/utils.js:14]: In that case we can define a type for the `keyType: 'alertname+namespace' | 'group_id'` to make this function more robust at least in compilation time as TypeScript will check that the correct values are passed as params

**Pull request merged**